script/intentfest/conversation.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+
+"""Simulate a conversation to check what the conversation would look like."""
+from __future__ import annotations
+
+import argparse
+import json
+import traceback
+from threading import Thread
+from time import sleep, time
+
+from watchdog.events import PatternMatchingEventHandler
+from watchdog.observers import Observer
+
+from .const import LANGUAGES
+from .sentenceParser import SentenceParser
+from .util import get_base_arg_parser
+
+
+def get_arguments() -> argparse.Namespace:
+    """Get parsed parameters passed in arguments."""
+    parser = get_base_arg_parser()
+    parser.add_argument(
+        "--language",
+        required=True,
+        type=str,
+        choices=LANGUAGES,
+        help="The language to validate.",
+    )
+    parser.add_argument(
+        "--no-response-debug",
+        action="store_true",
+        help="If set, there will be no response debug data.",
+    )
+
+    return parser.parse_args()
+
+
+class SentencesFilesWatcher(PatternMatchingEventHandler):
+    def __init__(self, callback_files_finished_changing, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.last_modification_time = None
+        self.running = False
+        self.callback_files_finished_changing = callback_files_finished_changing
+
+    def was_change(self):
+        if not self.running:
+            self.running = True
+            while time() - self.last_modification_time <= 0.3:
+                sleep(0.1)
+            self.running = False
```

Comment:
Use [`Timer`](https://docs.python.org/3/library/threading.html#threading.Timer). Cancel and create a new timer on every incoming event.